tests/foreman/ui/test_operatingsystem.py

```diff
@@ -118,49 +81,71 @@ def test_positive_end_to_end(session):
         assert len(os['parameters']['os_params']) == 1
         assert os['parameters']['os_params'][0]['name'] == param_name
         assert os['parameters']['os_params'][0]['value'] == param_value
+        template_name = gen_string('alpha')
         new_description = gen_string('alpha')
+        new_name = gen_string('alpha')
+        new_major_version = gen_string('numeric', 2)
+        new_minor_version = gen_string('numeric', 2)
+        new_family = 'Red Hat CoreOS'
+        new_hash = HASH_TYPE['sha512']
+        new_architecture = target_sat.api.Architecture().create()
+        new_ptable = target_sat.api.PartitionTable(
+            organization=[module_org], location=[module_location], os_family='Redhat'
+        ).create()
+        new_medium = target_sat.api.Media(
+            organization=[module_org], location=[module_location]
+        ).create()
+        new_param_name = gen_string('alpha')
+        new_param_value = gen_string('alpha')
+        session.provisioningtemplate.create(
+            {
+                'template.name': template_name,
+                'template.default': True,
+                'type.snippet': False,
+                'template.template_editor.editor': gen_string('alpha'),
+                'type.template_type': 'Provisioning template',
+                'association.applicable_os.assigned': [os['operating_system']['description']],
+                'organizations.resources.assigned': [module_org.name],
+                'locations.resources.assigned': [module_location.name],
+            }
+        )
+        # Update
         session.operatingsystem.update(
-            description, {'operating_system.description': new_description}
+            description,
+            {
+                'operating_system.name': new_name,
+                'templates.resources': {'Provisioning template': template_name},
+                'operating_system.description': new_description,
+                'operating_system.major': new_major_version,
+                'operating_system.minor': new_minor_version,
+                'operating_system.family': new_family,
+                'operating_system.password_hash': new_hash,
+                'operating_system.architectures.assigned': [new_architecture.name],
+                'partition_table.resources.assigned': [new_ptable.name],
+                'installation_media.resources.assigned': [new_medium.name],
+                'parameters.os_params': {'name': new_param_name, 'value': new_param_value},
```

Comment:
I think in update we should cover updating old `param_name`'s value, instead of adding new param, right?

Reply:
Updated